components/psst/browser/core/psst_consent_data.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_PSST_BROWSER_CORE_PSST_CONSENT_DATA_H_
+#define BRAVE_COMPONENTS_PSST_BROWSER_CORE_PSST_CONSENT_DATA_H_
+
+#include <optional>
+#include <string>
+
+#include "base/functional/callback.h"
+#include "base/values.h"
+#include "url/origin.h"
+
+namespace psst {
+
+using ConsentCallback =
+    base::OnceCallback<void(const base::Value::List disabled_checks)>;
+
+// Represents all the data required to display the consent dialog to the user.
+// This includes the information to be shown, as well as configuration for user
+// interaction and capturing their consent response.
+struct PsstConsentData {
```

Comment:
why have two separate structs `PsstConsentData` and `PsstSettings/Metadata`? Can't this be the one struct? And you need to remove callback from this struct. Just make it the data storage and pass UI-specific callback separately.